android/src/main/java/com/supersami/foregroundservice/ForegroundService.java

```diff
@@ -113,35 +109,71 @@ private boolean startService(Bundle notificationConfig) {
         }
     }
 
+    private int getServiceTypeForAndroid10(String customServiceType) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            switch (customServiceType) {
+                case "camera":
+                    return 8; // ServiceInfo.FOREGROUND_SERVICE_TYPE_CAMERA
+                case "connectedDevice":
+                    return 32; // ServiceInfo.FOREGROUND_SERVICE_TYPE_CONNECTED_DEVICE
+                case "dataSync":
+                    return 16; // ServiceInfo.FOREGROUND_SERVICE_TYPE_DATA_SYNC
+                case "health":
+                    return 64; // ServiceInfo.FOREGROUND_SERVICE_TYPE_HEALTH
+                case "location":
+                    return 1; // ServiceInfo.FOREGROUND_SERVICE_TYPE_LOCATION
+                case "mediaPlayback":
+                    return 2; // ServiceInfo.FOREGROUND_SERVICE_TYPE_MEDIA_PLAYBACK
+                case "mediaProjection":
+                    return 4; // ServiceInfo.FOREGROUND_SERVICE_TYPE_MEDIA_PROJECTION
+                case "microphone":
+                    return 128; // ServiceInfo.FOREGROUND_SERVICE_TYPE_MICROPHONE
+                case "phoneCall":
+                    return 256; // ServiceInfo.FOREGROUND_SERVICE_TYPE_PHONE_CALL
+                case "remoteMessaging":
+                    return 1024; // ServiceInfo.FOREGROUND_SERVICE_TYPE_REMOTE_MESSAGING
+                case "shortService":
+                    return 2048; // ServiceInfo.FOREGROUND_SERVICE_TYPE_SHORT_SERVICE
+                case "specialUse":
+                    return 4096; // ServiceInfo.FOREGROUND_SERVICE_TYPE_SPECIAL_USE
+                case "systemExempted":
+                    return 8192; // ServiceInfo.FOREGROUND_SERVICE_TYPE_SYSTEM_EXEMPTED
+                default:
+                    return 1; // Default to location
+            }
+        }
+        return 0; // This won't be used for Android < 10
+    }
+
     private int mapServiceType(String customServiceType) {
+        // Use direct integer constants instead of ServiceInfo constants
         switch (customServiceType) {
             case "camera":
-                return ServiceInfo.FOREGROUND_SERVICE_TYPE_CAMERA;
+                return 8; // ServiceInfo.FOREGROUND_SERVICE_TYPE_CAMERA
```

Comment:
@Raja0sama - Are these integer constants wrong? Our build crashes with this release with cryptic errors and I figured it was mismatch magic numbers.

I look up https://developer.android.com/reference/android/content/pm/ServiceInfo#FOREGROUND_SERVICE_TYPE_CAMERA and its 64

FOREGROUND_SERVICE_TYPE_CONNECTED_DEVICE - 16
FOREGROUND_SERVICE_TYPE_DATA_SYNC - 1
FOREGROUND_SERVICE_TYPE_HEALTH - 256

etc

They don't seem to match up. Am I looking at wong area?